src/FrontDoor/FrontDoor/Cmdlets/SetAzureRmFrontDoor.cs

```diff
@@ -104,23 +108,36 @@ public class SetAzureRmFrontDoor : AzureFrontDoorCmdletBase
         /// Whether to enable use of this rule.
         /// </summary>
         [Parameter(Mandatory = false, HelpMessage = "Operational status of the Front Door load balancer. Default value is Enabled")]
+        [PSArgumentCompleter("Enabled", "Disabled")]
         public PSEnabledState EnabledState { get; set; }
 
         /// <summary>
         /// Whether to enforce certificate name check on HTTPS requests to all backend pools. No effect on non-HTTPS requests.
         /// </summary>
-        [Parameter(Mandatory = false, HelpMessage = "Whether to disable certificate name check on HTTPS requests to all backend pools. No effect on non-HTTPS requests.")]
+        [Parameter(Mandatory = true, ParameterSetName = FieldsWithCertificateNameCheckParameterSet, HelpMessage = "Whether to disable certificate name check on HTTPS requests to all backend pools. No effect on non-HTTPS requests.")]
```

Comment:
Same here about shoundn't specify `mandatory`

Reply:
The switch parameter here has to be mandatory in the specified parameter sets. If I set it to false, if a user attempts to call Set-AzFrontDoor without it, PS returns and error "Parameter Set cannot be resolved using the specified named parameters". The parameter sets "FieldsParamterSet", "ObjectParameterSet", and "ResourceIdParameterSet", cover the case if DisableCertificateName check switch parameter (and BackendPoolsSetting) are both not present.